### ISSUE TITLE
⌨ Notification dialog missing title + description association

### ISSUE THREAD

**Issue by esizer** [GCTC-NTGC/gc-digital-talent#10850]
## ⌨ Accessibility Issue

The dialog containing the notifications is a custom implementation. Due to that, we missed the title and description since our existing component automatically adds those for us.

## 🦋 Expected Behaviour

Notification dialog title and description are properly associated with the root dialog role.

## 📋 Steps to Reproduce

1. Do a dev build `pnpm run dev`
2. Login as any user
3. Open the notification dialog
4. Observe the errors/warnings in the console related to the dialog

## 📸 Screenshot

![345136400-3866350e-f24c-4815-9afa-fc123cd2c608](https://github.com/GCTC-NTGC/gc-digital-talent/assets/4127998/36c788dd-fafb-4efb-a363-19db989c6735)

## 🙋‍♀️ Proposed Solution

Wrap the title and descriptions in the appropriate primitives.

## ✅ Acceptance Criteria

- [ ] Notification dialog title + description are associated to the root dialog
- [ ] No errors or warnings
